packages/math/range.ts

```diff
@@ -0,0 +1,36 @@
+// Given two ranges, return if the two ranges overlap with each other
+// e.g. [1, 3] overlaps with [2, 4] while [1, 3] does not overlap with [4, 5]
+export const rangesOverlap = (
+  [a0, a1]: [number, number],
```

Comment:
do you think we should have a branded type for a range as well?, we have [a, b], (a, b], (a, b), i'd say it's at least necessary to consider [a, b].

Reply:
Done!